### PR TITLE
Updated the Cosmosdb  module

### PR DESCRIPTION
**Pull request by mahhanifi**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------

- The Cosmosdb module is updated to include the networking aspect 
- separated the test variables in a separate file
- An issue is identified in the integration test, the Issue is created and those lines of code are commented out for the time being.
- Read me is updated

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

### CODE REVIEW

**Comment by helayoty** [comment]
#403

File: test-harness/terratest-extensions/modules/azure/cosmos.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package azure
```
Comment (nmiodice) [inline test-harness/terratest-extensions/modules/azure/cosmos.go:1]: So happy to see this!

File: test-harness/terratest-extensions/modules/azure/datafactory.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
+package azure
```
Comment (nmiodice) [inline test-harness/terratest-extensions/modules/azure/datafactory.go:1]: Perhaps from a different PR? Can you remove this, unless it needs to be here for some reason?

File: infra/modules/providers/azure/cosmosdb/tests/integration/cosmos.go [hunk 3]
```diff
@@ -50,20 +47,20 @@ func getModuleOutputs(output infratests.TerraformOutput, outputName string) (id
 }
 
 // validateServiceResponse - Attempt to perform a HTTP request to the live endpoint.
-func validateServiceResponse(t *testing.T, output infratests.TerraformOutput, outputName string) {
-	_, endpoint, _, _ := getModuleOutputs(output, outputName)
-	statusCode, _ := httpClient.HttpGet(t, endpoint)
+// func validateServiceResponse(t *testing.T, output integration.TerraformOutput, outputName string) {
+// 	_, endpoint, _, _ := getModuleOutputs(output, outputName)
+// 	statusCode, _ := httpClient.HttpGet(t, endpoint)
```
Comment (nmiodice) [inline infra/modules/providers/azure/cosmosdb/tests/integration/cosmos.go:52]: Looking at the code, I think adding a third argument of nil will do the trick. This argument refers to the TlsConfig. If nil, the default is used
 

https://github.com/gruntwork-io/terratest/blob/master/modules/http-helper/http_helper.go#L21

Is it possible to attempt to fix this? If it gives you trouble, I'm OK with commenting it for now.

File: infra/modules/providers/azure/cosmosdb/variables.tf [hunk 4]
```diff
@@ -48,4 +48,10 @@ variable "databases" {
     throughput = number
   }))
   default = []
+}
+
+# VNET Restriction (optional)
```
Comment (nmiodice) [inline infra/modules/providers/azure/cosmosdb/variables.tf:53]: If optional, do we need a default variable?

Comment (mahhanifi) [reply infra/modules/providers/azure/cosmosdb/variables.tf:53]: added `"10.2.0.0/16"` as default

**Pull request merged**